Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -6,12 +6,25 @@
 
 import Foundation
 
+/// In case that you are not able to use `DDURLSessionDelegate` or its subclasses in your `URLSession` instance,
+/// you can use `DDURLSessionDelegateProviding` protocol.
+/// Make sure that your type implements the same `URLSessionTaskDelegate` methods as `DDURLSessionDelegate` does.
+/// Otherwise your `URLSession` will not be instrumented automatically.
+@objc
+public protocol DDURLSessionDelegateProviding {
+    var delegate: DDURLSessionDelegate { get }
```

Comment:
tbh i still have doubts about this protocol

1. we can kind of hide it with discouraging comments while keeping it `public`
2. we don't have to introduce this protocol. i can still go with the other (`sharedInterceptor`) solution, its only problem was non-user-facing testability issues.
3. if we go with adding more `var ***delegate` then every new addition will be a breaking change -> major version update. if this protocol wasn't public, we could make those changes in minor/patch version updates.
4. if we go with specialised `xxxDelegateProviding` protocols then in the future we will have to make sure that existing protocols keep working, this may cause problems for our development

in any case, we communicate via comments which is not the ideal way. our users can end up with wrong implementation while the code still compiles successfully.

wdyt?

Reply:
My point is to also make something usable to this small number of users who currently report _"I can't subclass `DDURLSessionDelegate`"_ problem. The composition-based approach proposed in this PR may work as a replacement of inheritance.

> in any case, we communicate via comments which is not the ideal way. our users can end up with wrong implementation while the code still compiles successfully.

Yes, but the alternative is to provide no instrumentation for users who can't subclass `DDURLSessionDelegate` or do it through swizzling, resulting with tracking every single request made from the app or its 3rd party frameworks.

Anyway, this discussion is far from the core topic - adding `XCFrameworks`. What I proposed is solving another problem as a bonus, but we can do it separate.